### PR TITLE
Vagrant cluster ip

### PR DESCRIPTION
**Pull request by granders**
Updated to use externally_routable_ip in a few places.

### CODE REVIEW

File: aws/aws-access-keys-commands [hunk 1]
```diff
@@ -1,6 +1,6 @@
 if [ -z "$AWS_IAM" ];then
     # Default amazon iam identity
-    AWS_IAM=ducktape-master
+    AWS_IAM=ducttape-master
```
Comment (ewencp) [inline aws/aws-access-keys-commands]: I think this reversion snuck in by accident?

File: muckrake/services/kafka_rest.py [hunk 2]
```diff
@@ -74,5 +74,5 @@ def clean_node(self, node):
         node.account.ssh("rm -rf /mnt/rest.properties %(path)s" % self.logs["rest_log"], allow_fail=True)
 
     def url(self, idx=1):
-        return "http://" + self.get_node(idx).account.hostname + ":" + str(self.port)
+        return "http://" + self.get_node(idx).account.externally_routable_ip + ":" + str(self.port)
```
Comment (ewencp) [inline muckrake/services/kafka_rest.py]: Hmm, I think this is where this split internal/external scheme becomes a bit annoying. For the benefit of keeping traffic internal to EC2, I think you want the default here to use the `hostname` field, and maybe use a flag to indicate when you want the external address. Defaulting to internal avoids going through the external IP address, which is potentially slower, costs $ for bandwidth, etc.

File: muckrake/services/schema_registry.py [hunk 3]
```diff
@@ -93,30 +93,29 @@ def get_master_node(self):
...
         # Return the node with this base_url
         for idx, node in enumerate(self.nodes, 1):
-            if self.url(idx).find(base_url) >= 0:
-                return self.get_node(idx)
+            if node.account.hostname == hostname:
+                return node
 
     def url(self, idx=1):
-        return "http://" + self.get_node(idx).account.hostname + ":" + str(self.port)
+        return "http://" + self.get_node(idx).account.externally_routable_ip + ":" + str(self.port)
```
Comment (ewencp) [inline muckrake/services/schema_registry.py]: Same deal with the default here. The externally routable one should be used only when absolutely required.

File: aws/aws-access-keys-commands (hunk 1 above)
Comment (granders) [inline aws/aws-access-keys-commands]: It's intentional though not pretty - the ducktape-master iam isn't properly set up

**Comment by granders** [comment]
@ewencp 
Made url return 'internal' url by default

File: aws/aws-access-keys-commands (hunk 1 above)
Comment (ewencp) [inline aws/aws-access-keys-commands]: Ah, I see. Makes sense since this is from our infrastructure, not a generic ducktape thing.

**Comment by ewencp** [comment]
+1

**Pull request merged**